horace_core/sqw/file_io/sqw_formats_factory.m

```diff
@@ -195,8 +197,15 @@
                 the_type = class(varargin{1});
                 if isa(varargin{1},'sqw')
                     sobj = varargin{1};
-                    header =sobj.header;
-                    if iscell(header)
+                    header =sobj.my_header();
+                    if isa(header, 'Experiment')
+                        if isempty(header.expdata)
+                            loader = obj.supported_accessors_{1};
+                            return;
+                        else
+                            header = header.expdata(1);
+                        end
+                    elseif iscell(header)
                         header = header{1};
```

Comment:
`sqw_formats` factory should not know anything about experiment and what is in the file. it should not rely on the header to identify this.